tpcx_bb/queries/q28/tpcx_bb_query_28_sql.py

```diff
@@ -36,34 +36,33 @@
 N_FEATURES = 2 ** 23  # Spark is doing 2^20
 ngram_range = (1, 2)
 lowercase = True
-preprocessor = None
+preprocessor = lambda s:s
 norm = None
 alternate_sign = False
 
 
 def gpu_hashing_vectorizer(x):
-    return cudf_hashing_vectorizer(
-        x,
-        n_features=N_FEATURES,
-        alternate_sign=alternate_sign,
-        ngram_range=ngram_range,
-        norm=norm,
-        preprocessor=preprocessor,
-        lowercase=lowercase,
-    )
+    vec = HashingVectorizer(n_features=N_FEATURES,
+                            alternate_sign=alternate_sign,
+                            ngram_range=ngram_range,
+                            norm=norm,
+                            lowercase=lowercase,
+                            preprocessor=preprocessor
+     )
+    return vec.fit_transform(x)
 
 
-def map_labels(inp):
-    # Note: This gets JIT compiled for use with
-    # cuDF's `applymap` function so dict is not
-    # used.
-    x = inp
-    if x == 1 or x == 2:
-        return 0
-    elif x == 3:
-        return 1
-    else:
-        return 2
+def map_labels(ser):
+    import cudf
+    output_ser = cudf.Series(cp.full(shape=len(ser),fill_value=2,dtype=cp.int32))
```

Comment:
This looks like a great solution to this pre-compiling problem with the source loader. I think @galipremsagar just implemented this same functionality natively in cuDF.

On the off chance we end up with subtle cupy/cudf interaction bugs as the nightlies get released, might be good to switch to the cuDF version.

Reply:
https://github.com/rapidsai/cudf/blob/19237a09fe9b80706ef463580c6a646f6f9ec2ae/python/cudf/cudf/core/column/column.py#L1906

Reply:
> https://github.com/rapidsai/cudf/blob/19237a09fe9b80706ef463580c6a646f6f9ec2ae/python/cudf/cudf/core/column/column.py#L1906

+1

Reply:
Resolved by https://github.com/rapidsai/tpcx-bb/pull/104/commits/02057c55cf89de3ecbd72c1c031c3f92257283a2